src/runtime/opencl.cpp

```diff
@@ -1082,6 +1053,38 @@ WEAK uintptr_t halide_opencl_get_cl_mem(void *user_context, halide_buffer_t *buf
     return (uintptr_t)buf->device;
 }
 
+WEAK int halide_opencl_device_crop(void *user_context,
+                                    const struct halide_buffer_t *src,
+                                    struct halide_buffer_t *dst) {
+    dst->device_interface = src->device_interface;
+    int err = 0;
+    int64_t offset = 0;
+    for (int i = 0; i < src->dimensions; i++) {
+        offset += (dst->dim[i].min - src->dim[i].min) * src->dim[i].stride;
+    }
+    offset *= src->type.bytes();
+    cl_buffer_region region = {offset, dst->size_in_bytes()};
+    dst->device = (uint64_t)clCreateSubBuffer((cl_mem)(src->device),
```

Comment:
Added a comment. That's the expected behavior - the strides don't change when you crop. Really I just want to bump the min pointer to the top-left of the crop region. This is a way to do that in opencl.